### PR TITLE
Aleph upgrade will throw an exception if any ES call returns a status code < 399

### PR DESCRIPTION
**Pull request by catileptic**

### CODE REVIEW

File: aleph/index/util.py [hunk 1]
```diff
@@ -56,6 +56,10 @@
 }
 
 
+class AlephOpertionalException(Exception):
```
Comment (stchris) [inline aleph/index/util.py:59]:
```suggestion
class AlephOperationalException(Exception):
```

File: aleph/index/util.py [hunk 2]
```diff
@@ -237,8 +241,7 @@ def _check_response(index, res):
     if res.get("status", 0) > 399 and not res.get("acknowledged"):
         error = res.get("error", {}).get("reason")
         log.error("Index [%s] error: %s", index, error)
-        return False
-    return True
+        raise AlephOpertionalException(f"Index {index} error: {error}")
```
Comment (stchris) [inline aleph/index/util.py:244]:
```suggestion
        raise AlephOperationalException(f"Index {index} error: {error}")
```

File: aleph/manage.py [hunk 3]
```diff
@@ -499,9 +500,12 @@ def publish(foreign_id):
 @cli.command()
 def upgrade():
     """Create or upgrade the search index and database."""
-    upgrade_system()
-    # update_roles()
-    # upgrade_collections()
+    try:
+        upgrade_system()
+        # update_roles()
+        # upgrade_collections()
```
Comment (stchris) [inline aleph/manage.py:506]:
```suggestion
        update_roles()
        upgrade_collections()
```

File: aleph/manage.py [hunk 4]
```diff
@@ -21,6 +21,7 @@
 from aleph.queues import get_active_dataset_status
 from aleph.index.admin import delete_index
 from aleph.index.entities import iter_proxies
+from aleph.index.util import AlephOpertionalException
```
Comment (stchris) [inline aleph/manage.py:24]:
```suggestion
from aleph.index.util import AlephOperationalException
```

File: aleph/manage.py [hunk 5]
```diff
@@ -499,9 +500,12 @@ def publish(foreign_id):
...
 def upgrade():
     """Create or upgrade the search index and database."""
-    upgrade_system()
-    # update_roles()
-    # upgrade_collections()
+    try:
+        upgrade_system()
+        # update_roles()
+        # upgrade_collections()
+    except AlephOpertionalException as e:
```
Comment (stchris) [inline aleph/manage.py:507]:
```suggestion
    except AlephOperationalException as e:
```

File: aleph/index/util.py [hunk 6]
```diff
@@ -288,24 +291,27 @@ def configure_index(index, mapping, settings):
...
+        log.info(f"[{index}] Current mappings: {config.get("mappings")}")
+        log.info(f"[{index}] New mappings: {mapping}")
         mapping = rewrite_mapping_safe(mapping, config.get("mappings"))
         res = es.indices.put_mapping(body=mapping, ignore=[400], **options)
-        if not _check_response(index, res):
-            return False
+        _check_response(index, res)
         res = es.indices.open(**options)
-        return True
+        _check_response(index, res)
```
Comment (stchris) [inline aleph/index/util.py:309]: I wonder if this style is worth the line count savings:

```python
_check_response(index, es.indices.open(**options))
```

File: aleph/manage.py (hunk 3 above)
Comment (catileptic) [reply aleph/manage.py:506]: These two lines were commented 4 years ago ([here](https://github.com/alephdata/aleph/commit/0350dabefedda6af8390ccf7574227118208bacf)). Do we want to bring them back?

File: aleph/index/util.py [hunk 7]
```diff
@@ -291,27 +291,24 @@ def configure_index(index, mapping, settings):
         }
         config = es.indices.get(index=index).get(index, {})
         settings.get("index").pop("number_of_shards")
-        log.info(f"[{index}] Current settings: {config.get("settings")}")
+        log.info(f"[{index}] Current settings.", settings=config.get("settings"))
```
Comment (stchris) [inline aleph/index/util.py:294]: Might as well put `index=index` as an argument to the logger here since we're logging structured data.

**Review by stchris — APPROVED** [review]
Nice work, thanks for this! 👏

**Pull request merged**